tests/e2e/chains/ethereum/contracts/hardhat.config.js

```diff
@@ -1,19 +1,40 @@
 require("@nomicfoundation/hardhat-toolbox");
+require("@nomicfoundation/hardhat-foundry");
 require("@openzeppelin/hardhat-upgrades");
+require("hardhat-contract-sizer");
 
 /**
  * @type import('hardhat/config').HardhatUserConfig
+ * @notice `contracts/App.sol` is compiled with different settings because AppV7's size exceeds the contract size limit if viaIR is enabled
  */
 module.exports = {
   solidity: {
-    version: "0.8.28",
-    settings: {
-      evmVersion: "cancun",
-      optimizer: {
-        enabled: true,
-        runs: 9_999_999
+    compilers: [
+      {
+        version: "0.8.28",
+        settings: {
+          evmVersion: "cancun",
+          viaIR: true,
+          optimizer: {
+            enabled: true,
+            runs: 9_999_999
+          }
+        }
       }
-    },
+    ],
+    overrides: {
+      "contracts/App.sol": {
+        version: "0.8.28",
+        settings: {
+          evmVersion: "cancun",
+          viaIR: false,
+          optimizer: {
+            enabled: true,
+            runs: 9_999_999
+          }
+        }
+      }
+    }
```

Comment:
Why do we have to disable `viaIR` for `contracts/App.sol`?

Reply:
the reason is here: https://github.com/datachainlab/cosmos-ethereum-ibc-lcp/pull/69/files/14f5babfe18c7411aab9863b24ee6cfa59ae20f9#diff-30726da73db15f2f0e70aef722159de99681d95f365191c93b2c635c350c74f6R8